From 5032d6c0ab4a8f031e4d8847c03576a93308dcfa Mon Sep 17 00:00:00 2001
From: Bram Moolenaar <Bram@vim.org>
Date: Wed, 22 May 2019 22:38:25 +0200
Subject: [PATCH 1/3] patch 8.1.1365: source command doesn't check for the
 sandbox

Problem: Source command doesn't check for the sandbox. (Armin Razmjou)
Solution: Check for the sandbox when sourcing a file.

Signed-off-by: Felix <google@ix5.org>
Change-Id: I10c69e1921a23209868a27f77999eaa966739a39
---
 src/getchar.c | 6 ++++++
 src/version.c | 2 ++
 2 files changed, 8 insertions(+)

diff --git a/src/getchar.c b/src/getchar.c
index 3657433a5..bd7b19c07 100644
--- a/src/getchar.c
+++ b/src/getchar.c
@@ -1419,6 +1419,12 @@ openscript(
 	EMSG(_(e_nesting));
 	return;
     }
+
+    // Disallow sourcing a file in the sandbox, the commands would be executed
+    // later, possibly outside of the sandbox.
+    if (check_secure())
+        return;
+
 #ifdef FEAT_EVAL
     if (ignore_script)
 	/* Not reading from script, also don't open one.  Warning message? */
diff --git a/src/version.c b/src/version.c
index 75bb16ab2..ad0ec3b78 100644
--- a/src/version.c
+++ b/src/version.c
@@ -764,6 +764,8 @@ static char *(features[]) =
 
 static int included_patches[] =
 {   /* Add new patch number below this line */
+/**/
+    1365,
 /**/
     133,
 /**/
-- 
2.17.1

